src/Microsoft.DocAsCode.Build.SchemaDriven/SchemaFragmentsIterator.cs

```diff
@@ -102,8 +102,15 @@ public SchemaFragmentsIterator(ISchemaFragmentsHandler handler)
                     {
                         if (item is YamlMappingNode mapNode)
                         {
-                            var opath = string.Format("{0}[{1}=\"{2}\"]", parentOPath, mergeKey, mapNode.Children[mergeKey].ToString());
-                            TraverseCore(item, fragments, schema.Items, opath, uid);
+                            if (mapNode.Children.ContainsKey(mergeKey))
+                            {
+                                var opath = string.Format("{0}[{1}=\"{2}\"]", parentOPath, mergeKey, mapNode.Children[mergeKey].ToString());
+                                TraverseCore(item, fragments, schema.Items, opath, uid);
+                            }
+                            else
+                            {
+                                Logger.LogError($"Cannot find merge key {mergeKey} in {mapNode.ToString()}");
```

Comment:
>LogError [](start = 39, length = 8)

@superyyrrzz@gmail.com 
When there is error, will the caller of SchemaFragmentsIterator terminate?

Reply:
No, only an error is logged.

Reply:
is it expected?

---
In reply to: [200884075](https://github.com/dotnet/docfx/pull/3019#discussion_r200884075) [](ancestors = 200884075)

Reply:
Seems SchemaFragmentsIterator should throw, while the call should catch and handle it. I'm not sure whether exception is needed here. Will talk to Tianqi about this.